File: readability.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

// Declaring the function which counts the letters (body of the function is below the main)
int count_letters(string txt);
int count_words(string txt);
int count_sent(string txt);

int main(void)
{
    // Function first takes string from user, stores it in "usertxt" and then takes it to the function count_letters
    // which returns the number of letters and stores it in a variable "letters"
    string usertxt = get_string("Text: ");
    
    int letters = count_letters(usertxt);
    int words = count_words(usertxt);
    int sent = count_sent(usertxt);
  
    float l = 100 * letters / words;
    float s = 100 * sent / words;
  
    float index = 0.0588 * l - 0.296 * s - 15.8;
 
    float ind = round(index);
  
  
    if(ind >= 1 && ind <= 15)
    {
        printf("Grade %.0f\n", ind);
    }
    else if (ind < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (ind >= 16)
    {
        printf("Grade 16+\n");
    }
}

// Defining the function
int count_letters(string txt)
{
    int sum = 0;

    // Loop that checks every character in a text given by the user (which is actually an array of chars) if it is a letter
    for (int i = 0; txt[i] != '\0'; i++)
    {
      if ((txt[i] >= 'a' && txt[i] <= 'z') || (txt[i] >= 'A' && txt[i] <= 'Z'))
      {
        sum++;
      }
    }

    return sum;
}

// Defining the function
int count_words(string txt)
{
    int sum = 0;

    // If a "i"th character is a non-space, add one word, THEN: increment the number of chars in array until you encounter space
    // By doing this in a loop you get what you want babyyyy
    for (int i = 0; txt[i] != '\0'; i++)
    {
      if (txt[i] != ' ')
      {
        sum++;

        for (int n = 0; txt[i] != ' '; n++)
        {
            if (txt[i] == '\0')
            {
                break;
            }
            i++;
        }
      }
    }
    return sum;

}

// Defining the function
int count_sent(string txt)
{
    int sum = 0;

    // Loop for sentences, only three conditions needed
    for (int i = 0; txt[i] != '\0'; i++)
    {
      if (txt[i] == '.' || txt[i] == '!' || txt[i] == '?')
      {
        sum++;
      }
    }
    return sum;
}

File: readability.py
from cs50 import get_string

def countLetters(text):

    number = 0

    # Lower each character to have shorter condition
    txt = text.lower()

    for i in range(len(txt)):
        if (txt[i] >= 'a' and txt[i] <= 'z'):
            number += 1

    return number


# ANALIZE IT CAREFULLY ONCE AGAIN + opisac dla siebie lepiej while loop
def countWords(text):
    number = 0
    i = 0 
    
    while i < len(text):
        number += 1

        # iterate through word until space or end of string
        # while not executes while expression between the parentheses is FALSE (so both have to be false for loop to go through)
        while not (i == len(text) or text[i] == ' '):
            i += 1

        # iterate through space to the next char
        i += 1

    return number

def countSentences(text):

    number = 0
    
    for i in range(len(text)):
        if text[i] == '.' or text[i] == '!' or text[i] == '?':
            number += 1
    
    return number




usertext = get_string("Text: ")

letters = countLetters(usertext);
words = countWords(usertext);
sentences = countSentences(usertext);

L = float(100 * letters/words)
S = float(100 * sentences / words)

index = round(float(0.0588 * L - 0.296 * S - 15.8))
if index >= 1 and index <= 15:
    print(f"Grade {index}")
elif index < 1:
    print("Before Grade 1")
elif index >= 16:
    print ("Grade 16+")


